pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -367,53 +367,60 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 
 			buildRun.Status.LatestTaskRunRef = &lastTaskRun.Name
 			buildRun.Status.StartTime = lastTaskRun.Status.StartTime
-
 			if lastTaskRun.Status.CompletionTime != nil && buildRun.Status.CompletionTime == nil {
 				buildRun.Status.CompletionTime = lastTaskRun.Status.CompletionTime
 
-				// Increase BuildRun count in metrics
-				buildmetrics.BuildRunCountInc(buildRun.Status.BuildSpec.StrategyRef.Name)
-
-				// buildrun established duration (time between the creation of the buildrun and the start of the buildrun)
-				buildmetrics.BuildRunEstablishObserve(
-					buildRun.Status.BuildSpec.StrategyRef.Name,
-					buildRun.Namespace,
-					buildRun.Status.StartTime.Time.Sub(buildRun.CreationTimestamp.Time),
-				)
-
-				// buildrun completetion duration (total time between the creation of the buildrun and the buildrun completion)
-				buildmetrics.BuildRunCompletionObserve(
-					buildRun.Status.BuildSpec.StrategyRef.Name,
-					buildRun.Namespace,
-					buildRun.Status.CompletionTime.Time.Sub(buildRun.CreationTimestamp.Time),
-				)
-
-				// buildrun ramp-up duration (time between buildrun creation and taskrun creation)
-				buildmetrics.BuildRunRampUpDurationObserve(
-					buildRun.Status.BuildSpec.StrategyRef.Name,
-					buildRun.Namespace,
-					lastTaskRun.CreationTimestamp.Time.Sub(buildRun.CreationTimestamp.Time),
-				)
-
-				// Look for the pod created by the taskrun
-				var pod = &corev1.Pod{}
-				if err := r.client.Get(ctx, types.NamespacedName{Namespace: request.Namespace, Name: lastTaskRun.Status.PodName}, pod); err == nil {
-					lastInitPodIdx := len(pod.Status.InitContainerStatuses) - 1
-					lastInitPod := pod.Status.InitContainerStatuses[lastInitPodIdx]
-
-					// taskrun ramp-up duration (time between taskrun creation and taskrun pod creation)
-					buildmetrics.TaskRunRampUpDurationObserve(
+				if buildRun.Status.BuildSpec.StrategyRef != nil {
```

Comment:
Hm, I think you are uncovering a bug in the `BuildSpec`. The `StrategyRef` is not optional and therefore should not be a reference - similar to `Source` and `Output` that are also mandatory. Can you change this in `build_types.go` and remove this nil check here again ?

Reply:
removed, thats a good point. Unit tests know nothing about the tags. Also, when you say "Can you change this in build_types.go" , what do you meant?

Reply:
I mean this loc: https://github.com/shipwright-io/build/blob/87b5d4c55f9b13d83fc6a8c4da34d81d3d346c73/pkg/apis/build/v1alpha1/build_types.go#L28. It should not be `*StrategyRef` but `StrategyRef`. But, I just looked around, there are more similar problems like the `BuildRef` in the build run. So, you can also leave it as is and I open an issue to separately address all this problems. If you quickly confirm, I will approve this PR and address the wrong references separately.

Reply:
Yes, pls open the issue first. I would like to have the discussion around the pointer or not, interesting.

Reply:
Issue opened: https://github.com/shipwright-io/build/issues/397

Let's provide feed back there. This PR is fine then.